client/gutenberg/editor/controller.js

```diff
@@ -62,6 +65,48 @@ function waitForSiteIdAndSelectedEditor( context ) {
 	} );
 }
 
+function handleJetpackSSO( context ) {
+	// If we are dealing with an Atomic or Jetpack site we need to make sure that Jetpack SSO
+	// has been handled before we load the editor in an iFrame in order to prevent any issues
+	// with 3rd party cookie setting being blocked by the browser
+	const state = context.store.getState();
+	const currentRoute = getCurrentRoute( state );
+	const { URL: selectedSiteUrl, domain: selectedSiteDomain, jetpack } = getSelectedSite( state );
+
+	if ( ! jetpack ) {
+		return;
+	}
+
+	const {
+		hostname: parentDomain,
+		protocol: parentProtocol,
+		port,
+		query: parentQuery,
+	} = urlLib.parse( window.location.href, true );
+	const parentPort = port ? `:${ port }` : '';
+
+	// check query params from parent frame to check if we have already redirected back from Jetpack auth.
+	if ( parentQuery.jetpackSSO ) {
+		// If successfully redirected save to session storage so we don't need to redirect on every editor load
+		sessionStorage.setItem( `calypsoify_${ selectedSiteDomain }_jetpackSSO`, 'true' );
+		return;
+	}
+
+	if ( sessionStorage.getItem( `calypsoify_${ selectedSiteDomain }_jetpackSSO` ) ) {
+		return;
+	}
+
+	// if site is not authenticated yet then redirect to that domain to auth
+	// and redirect back here with jetpackSSO param set
+	parentQuery.jetpackSSO = 'true';
+	const returnURL = encodeURIComponent(
+		`${ parentProtocol }//${ parentDomain }${ parentPort }${ currentRoute }?${ stringify(
+			parentQuery
+		) }`
+	);
+	window.location.href = `${ selectedSiteUrl }/wp-login.php?redirect_to=${ returnURL }`;
```

Comment:
Calling `page.redirect` might be a bit nicer inside a page.js handler. In the end, it will do the same thing for an external URL.

Reply:
I ended up using `window.location.replace` because `page.redirect` was triggering this error:

<img width="1422" alt="Screen Shot 2019-08-02 at 14 40 56" src="https://user-images.githubusercontent.com/1233880/62347118-9f2d0c80-b533-11e9-88d8-bc0540ef7aba.png">